pkg/virtctl/expose/expose_test.go

```diff
@@ -308,6 +311,25 @@ var _ = Describe("Expose", func() {
 					Expect(cmd()).To(HaveOccurred(), "should fail on an invalid IP family")
 
 				})
+
+				table.DescribeTable("should select a valid default for the IPFamilyPolicy attribute, based on the provided IPFamilies", func(ipFamilies ...string) {
+					ipFamilyStr := strings.Join(ipFamilies, ",")
+					cmd := tests.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
+						"--port", "9999", "--target-port", "http", "--ip-family", ipFamilyStr)
+					Expect(cmd()).To(Succeed(), "should succeed with the default IP family policy")
+					Expect(obtainedService).NotTo(BeNil())
+
+					var expectedIPFamily *k8sv1.IPFamilyPolicyType
+					if len(ipFamilies) > 1 {
+						polType := k8sv1.IPFamilyPolicyPreferDualStack
+						expectedIPFamily = &polType
+					}
+					Expect(obtainedService.Spec.IPFamilyPolicy).To(Equal(expectedIPFamily))
+				},
+					table.Entry("a single IPv4 IPFamily", k8sv1.IPv4Protocol),
+					table.Entry("a single IPv6 IPFamily", k8sv1.IPv6Protocol),
+					table.Entry("a list of IPv4, IPv6 IPFamilies", k8sv1.IPv4Protocol, k8sv1.IPv6Protocol),
+					table.Entry("a list of IPv6, IPv4 IPFamilies", k8sv1.IPv6Protocol, k8sv1.IPv4Protocol))
```

Comment:
Put all the expectations at the test table don't put product code on the test like 

```golang
if len(ipFamilies) > 1 {
    polType := k8sv1.IPFamilyPolicyPreferDualStack
    expectedIPFamily = &polType
}
```

```suggestion
					table.Entry("a single IPv4 IPFamily", k8sv1.IPv4Protocol, ""),
					table.Entry("a single IPv6 IPFamily", k8sv1.IPv6Protocol, ""),
					table.Entry("a list of IPv4, IPv6 IPFamilies", k8sv1.IPv4Protocol, k8sv1.IPv6Protocol, k8sv1.IPFamilyPolicyPreferDualStack),
					table.Entry("a list of IPv6, IPv4 IPFamilies", k8sv1.IPv6Protocol, k8sv1.IPv4Protocol, k8sv1.IPFamilyPolicyPreferDualStack))
```

Reply:
Adapted the provided example (that doesn't compile). Hope it suits you.